### PR TITLE
Added restyler for https://github.com/nicklockwood/SwiftFormat

### PR DESCRIPTION
**Pull request by jacobcxdev**
See https://github.com/nicklockwood/SwiftFormat for more information.

### CODE REVIEW

File: swiftformat/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# ================================
+FROM swift:latest as build
+
+LABEL maintainer="JacobCXDev <jacobcxdev@gmail.com>"
+
+# Setup a build area
+WORKDIR /build
+
+# Clone and build SwiftFormat
+RUN git clone https://github.com/nicklockwood/SwiftFormat .
```
Comment (pbrisbin) [inline swiftformat/Dockerfile:12]: Can we do something like this, to make the version we're building explicit?

```suggestion
ENV SWIFT_VERSION 0.49.3
RUN git clone https://github.com/nicklockwood/SwiftFormat .
RUN git checkout "$SWIFT_VERSION"
```

There's probably an efficient way to shallow-clone the tag only, but I'd have to figure that out.

**Review by pbrisbin — CHANGES_REQUESTED** [review]
Thanks for the submission!

I'm going to request that you add some tests before I merge this. It's important to make sure things are (and continue to be) functional given that I'm not usually familiar with all the auto-formatters present as Restylers. Any other Restyler's `info.yaml` can be used as an example.

I apologize that this wasn't clearer up front; I'll update the wiki to indicate the importance of tests with these submissions.

If you run into any trouble, let me know.

File: swiftformat/Dockerfile (hunk 1 above)
Comment (jacobcxdev) [reply swiftformat/Dockerfile:12]: > I'm going to request that you add some tests before I merge this. It's important to make sure things are (and continue to be) functional given that I'm not usually familiar with all the auto-formatters present as Restylers. Any other Restyler's `info.yaml` can be used as an example.

I was planning on adding tests, but the formatter is configurable via the existence of a `.swiftformat` file (see https://github.com/nicklockwood/SwiftFormat#config-file); might this not interfere with any hard-coded tests, or am I misunderstanding how they work?

**Comment by pbrisbin** [comment]
:wave: Hi @jacobcxdev are you planning on coming back to this PR, or should I close it as stale?

**Comment by jacobcxdev** [comment]
> :wave: Hi @jacobcxdev are you planning on coming back to this PR, or should I close it as stale?

Hey, sorry for the radio silence; I'm currently sitting exams, so this will have to take the back seat for a while. That being said, I would certainly like to complete the PR at some point in the future!

**Comment by pbrisbin** [comment]
OK, no worries, I'll check back in again in a while. Good luck!

**Comment by pbrisbin** [comment]
@jacobcxdev friendly ping on this :)

**Comment by pbrisbin** [comment]
Closing as Stale. Feel free to re-open if you can get back to this. Hope the exams went well!

**Pull request closed without merging**